pkg/loki/modules.go

```diff
@@ -364,6 +364,7 @@ func (t *Loki) initQuerier() (services.Service, error) {
 
 	httpMiddleware := middleware.Merge(
 		httpreq.ExtractQueryMetricsMiddleware(),
+		serverutil.CacheGenNumberHeaderSetterMiddleware(t.cacheGenerationLoader),
```

Comment:
we can add a check for `supportIndexDeleteRequest` to see if we need to add this middleware or not?